builtin/providers/azurerm/resource_arm_resource_group.go

```diff
@@ -0,0 +1,162 @@
+// documented in the ARM REST API guide: http://bit.ly/1NEXclG
+func validateArmResourceGroupName(v interface{}, k string) (ws []string, es []error) {
+	value := v.(string)
+
+	if len(value) > 80 {
+		es = append(es, fmt.Errorf("%q may not exceed 80 characters in length", k))
+	}
+
+	if strings.HasSuffix(value, ".") {
+		es = append(es, fmt.Errorf("%q may not end with a period", k))
+	}
+
+	if matched := regexp.MustCompile(`[\(\)\.a-zA-Z0-9_-]`).Match([]byte(value)); !matched {
+		es = append(es, fmt.Errorf("%q may only contain alphanumeric characters, dash, underscores, parentheses and periods", k))
+	}
+
+	return
+}
+
+// resourceArmResourceGroupCreate goes ahead and creates the specified ARM resource group.
+func resourceArmResourceGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	resGroupClient := client.resourceGroupClient
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+
+	log.Printf("[INFO] Issuing Azure ARM creation request for resource group '%s'.", name)
+
+	rg := resources.ResourceGroup{
+		Name:     &name,
+		Location: &location,
+	}
+
+	_, err := resGroupClient.CreateOrUpdate(name, rg)
+	if err != nil {
+		return fmt.Errorf("Error issuing Azure ARM create request for resource group '%s': %s", name, err)
+	}
+
+	d.SetId(*rg.Name)
+
+	// Wait for the resource group to become available
+	// TODO(jen20): Is there any need for this?
+	log.Printf("[DEBUG] Waiting for Resource Group (%s) to become available", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Accepted"},
+		Target:  "Succeeded",
+		Refresh: resourceGroupStateRefreshFunc(client, d.Id()),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Resource Group (%s) to become available: %s", d.Id(), err)
+	}
+
+	return resourceArmResourceGroupRead(d, meta)
+}
+
+// resourceArmResourceGroupRead goes ahead and reads the state of the corresponding ARM resource group.
+func resourceArmResourceGroupRead(d *schema.ResourceData, meta interface{}) error {
+	resGroupClient := meta.(*ArmClient).resourceGroupClient
+
+	name := d.Id()
+	log.Printf("[INFO] Issuing read request to Azure ARM for resource group '%s'.", name)
+
+	res, err := resGroupClient.Get(name)
+	if err != nil {
+		return fmt.Errorf("Error issuing read request to Azure ARM for resource group '%s': %s", name, err)
+	}
+
+	d.Set("name", *res.Name)
+	d.Set("location", *res.Location)
+
+	return nil
+}
+
+// resourceArmResourceGroupExists goes ahead and checks for the existence of the correspoding ARM resource group.
+func resourceArmResourceGroupExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	resGroupClient := meta.(*ArmClient).resourceGroupClient
+
+	name := d.Id()
+
+	resp, err := resGroupClient.CheckExistence(name)
+	if err != nil {
+		// TODO(aznashwan): implement some error switching helpers in the SDK
```

Comment:
I'm glad you did not remove this TODO, it just reminded me I wanted to talk to the SDK guys about just such and error-management package...